subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java

```diff
@@ -120,6 +121,17 @@ public interface LongRunningOperation {
      */
     LongRunningOperation addJvmArguments(String... jvmArguments);
 
+
+    /**
+     * Sets additional system properties for the build.
+     *
+     * @param systemProperties the system properties add to the Gradle process
+     * @return this
+     * @since 7.6
+     */
+    @Incubating
+    LongRunningOperation withSystemProperties(Map<String, String> systemProperties);
```

Comment:
IIUC, these control the set of system properties to propagate to the build on top of "immutable" system properties that we always propagate.  

`withSystemProperties` can imply that these are additive, so users might expect this to have two system properties and not just the last one:
```
launcher.withSystemProperties([foo: "bar"]).withSystemProperties([baz: "bar"])
```

We should clarify in the docs here if it's additive or not and mention something about which immutable system properties will always propagate to the build.

Reply:
I've added the requested documentation.